port/esp32/adapter/src/clock.c

```diff
@@ -43,12 +45,8 @@ oc_clock_time(void)
 oc_clock_time_t
 oc_clock_time_monotonic(void)
 {
-  struct timespec t;
-  if (clock_gettime(CLOCK_MONOTONIC, &t) == -1) {
-    return -1;
-  }
-  return (oc_clock_time_t)t.tv_sec * OC_CLOCK_SECOND +
-         (oc_clock_time_t)ceil((double)t.tv_nsec / (1.e09 / OC_CLOCK_SECOND));
+  // return esp_timer_get_time returns microseconds from the boot time
```

Comment:
`OC_CLOCK_SECOND` might not be microseconds, so you need to do conversion to oc_clock_time_t

Reply:
fixed